relay-server/src/utils/memory.rs

```diff
@@ -0,0 +1,307 @@
+use crate::statsd::RelayGauges;
+use arc_swap::ArcSwap;
+use relay_config::Config;
+use relay_statsd::metric;
+use std::fmt;
+use std::fmt::Formatter;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::{Arc, Mutex};
+use std::time::Instant;
+use sysinfo::{MemoryRefreshKind, System};
+
+/// Count after which the [`MemoryStat`] data will be refreshed.
+const UPDATE_TIME_THRESHOLD_MS: u64 = 100;
```

Comment:
Should we make this configurable?

Reply:
We could do that, I could put it under the health key.